src/checkqueue.h

```diff
@@ -109,7 +109,7 @@ class CCheckQueue
                 //   all workers finish approximately simultaneously.
                 // * Try to account for idle jobs which will instantly start helping.
                 // * Don't do batches smaller than 1 (duh), or larger than nBatchSize.
-                nNow = std::max(1U, std::min(nBatchSize, (unsigned int)queue.size() / (nTotal + nIdle + 1)));
+                nNow = std::clamp<unsigned int>(queue.size() / (nTotal + nIdle + 1), 1, nBatchSize);
```

Comment:
Not sure if this can ever happen, but when `nBatchSize==0` this will now have undefined behavior. Previously the result used to be 1.

Reply:
Right. Would `assert(nBatchSize >= 1);` in the `CCheckQueue` constructor be enough?

Reply:
I'd put it right above the clamp

Reply:
I thought again, and put the assertion into the constructor because it is constructor's job to establish safe invariants for a class instance.